diff --git a/include/lzo/lzodefs.h b/include/lzo/lzodefs.h
index ec02624..06d99d3 100644
--- a/include/lzo/lzodefs.h
+++ b/include/lzo/lzodefs.h
@@ -824,6 +824,9 @@
 #elif (LZO_ARCH_CRAY_MPP) && (defined(_CRAYT3D) || defined(_CRAYT3E))
 #  define LZO_ARCH_ALPHA            1
 #  define LZO_INFO_ARCH             "alpha"
+#elif defined(__sw_64__)
+#  define LZO_ARCH_SW_64            1
+#  define LZO_INFO_ARCH             "sw_64"
 #elif defined(__amd64__) || defined(__x86_64__) || defined(_M_AMD64)
 #  define LZO_ARCH_AMD64            1
 #  define LZO_INFO_ARCH             "amd64"
@@ -2287,6 +2290,8 @@ LZO_COMPILE_TIME_ASSERT_HEADER(LZO_SIZEOF_LONG == sizeof(long))
 #if !defined(LZO_WORDSIZE)
 #if (LZO_ARCH_ALPHA)
 #  define LZO_WORDSIZE              8
+#elif (LZO_ARCH_SW_64)
+#  define LZO_WORDSIZE              8
 #elif (LZO_ARCH_AMD64)
 #  define LZO_WORDSIZE              8
 #elif (LZO_ARCH_ARM64)
@@ -2435,7 +2440,7 @@ LZO_COMPILE_TIME_ASSERT_HEADER(LZO_SIZEOF_PTRDIFF_T == sizeof(ptrdiff_t))
 #  define LZO_ABI_BIG_ENDIAN        1
 #elif (LZO_ARCH_IA64) && (LZO_OS_POSIX_LINUX || LZO_OS_WIN64)
 #  define LZO_ABI_LITTLE_ENDIAN     1
-#elif (LZO_ARCH_ALPHA || LZO_ARCH_AMD64 || LZO_ARCH_BLACKFIN || LZO_ARCH_CRIS || LZO_ARCH_I086 || LZO_ARCH_I386 || LZO_ARCH_MSP430 || LZO_ARCH_RISCV)
+#elif (LZO_ARCH_ALPHA || LZO_ARCH_SW_64 || LZO_ARCH_AMD64 || LZO_ARCH_BLACKFIN || LZO_ARCH_CRIS || LZO_ARCH_I086 || LZO_ARCH_I386 || LZO_ARCH_MSP430 || LZO_ARCH_RISCV)
 #  define LZO_ABI_LITTLE_ENDIAN     1
 #elif (LZO_ARCH_AVR32 || LZO_ARCH_M68K || LZO_ARCH_S390 || LZO_ARCH_SPU)
 #  define LZO_ABI_BIG_ENDIAN        1
@@ -2594,6 +2599,8 @@ LZO_COMPILE_TIME_ASSERT_HEADER(LZO_SIZEOF_PTRDIFF_T == sizeof(ptrdiff_t))
 #endif
 #if (LZO_ARCH_ALPHA)
 #  define LZO_OPT_AVOID_UINT_INDEX          1
+#elif (LZO_ARCH_SW_64)
+#  define LZO_OPT_AVOID_INT_INDEX           1
 #elif (LZO_ARCH_AMD64)
 #  define LZO_OPT_AVOID_INT_INDEX           1
 #  define LZO_OPT_AVOID_UINT_INDEX          1
diff --git a/minilzo/minilzo.c b/minilzo/minilzo.c
index 30948e7..b2fc80a 100644
--- a/minilzo/minilzo.c
+++ b/minilzo/minilzo.c
@@ -844,6 +844,9 @@
 #elif (LZO_ARCH_CRAY_MPP) && (defined(_CRAYT3D) || defined(_CRAYT3E))
 #  define LZO_ARCH_ALPHA            1
 #  define LZO_INFO_ARCH             "alpha"
+#elif defined(__sw_64__)
+#  define LZO_ARCH_SW_64            1
+#  define LZO_INFO_ARCH             "sw_64"
 #elif defined(__amd64__) || defined(__x86_64__) || defined(_M_AMD64)
 #  define LZO_ARCH_AMD64            1
 #  define LZO_INFO_ARCH             "amd64"
@@ -2307,6 +2310,8 @@ LZO_COMPILE_TIME_ASSERT_HEADER(LZO_SIZEOF_LONG == sizeof(long))
 #if !defined(LZO_WORDSIZE)
 #if (LZO_ARCH_ALPHA)
 #  define LZO_WORDSIZE              8
+#elif (LZO_ARCH_SW_64)
+#  define LZO_WORDSIZE              8
 #elif (LZO_ARCH_AMD64)
 #  define LZO_WORDSIZE              8
 #elif (LZO_ARCH_ARM64)
@@ -2455,7 +2460,7 @@ LZO_COMPILE_TIME_ASSERT_HEADER(LZO_SIZEOF_PTRDIFF_T == sizeof(ptrdiff_t))
 #  define LZO_ABI_BIG_ENDIAN        1
 #elif (LZO_ARCH_IA64) && (LZO_OS_POSIX_LINUX || LZO_OS_WIN64)
 #  define LZO_ABI_LITTLE_ENDIAN     1
-#elif (LZO_ARCH_ALPHA || LZO_ARCH_AMD64 || LZO_ARCH_BLACKFIN || LZO_ARCH_CRIS || LZO_ARCH_I086 || LZO_ARCH_I386 || LZO_ARCH_MSP430 || LZO_ARCH_RISCV)
+#elif (LZO_ARCH_ALPHA || LZO_ARCH_SW_64 || LZO_ARCH_AMD64 || LZO_ARCH_BLACKFIN || LZO_ARCH_CRIS || LZO_ARCH_I086 || LZO_ARCH_I386 || LZO_ARCH_MSP430 || LZO_ARCH_RISCV)
 #  define LZO_ABI_LITTLE_ENDIAN     1
 #elif (LZO_ARCH_AVR32 || LZO_ARCH_M68K || LZO_ARCH_S390 || LZO_ARCH_SPU)
 #  define LZO_ABI_BIG_ENDIAN        1
@@ -2614,6 +2619,8 @@ LZO_COMPILE_TIME_ASSERT_HEADER(LZO_SIZEOF_PTRDIFF_T == sizeof(ptrdiff_t))
 #endif
 #if (LZO_ARCH_ALPHA)
 #  define LZO_OPT_AVOID_UINT_INDEX          1
+#elif (LZO_ARCH_SW_64)
+#  define LZO_OPT_AVOID_INT_INDEX           1
 #elif (LZO_ARCH_AMD64)
 #  define LZO_OPT_AVOID_INT_INDEX           1
 #  define LZO_OPT_AVOID_UINT_INDEX          1
